cmd/thanos/sidecar.go

```diff
@@ -139,6 +148,12 @@ func runSidecar(
 				}
 			}
 
+			// When the heartbeat to Prometheus fails, the sidecar is marked as not ready.
+			// But after `sinceLastSuccessfulHeartbeatLimit` duration of consequential fails it's marked also not healthy,
+			// so the orchestrator (if any) can try restarting it if it would help.
+			sinceLastSuccessfulHeartbeat := 0 * time.Minute
```

Comment:
Thanks for those variables and comments. Super clear now :+1: 

However, are we sure about the value? If we cannot connect to Prometheus, we cannot use remote read either so I guess it make sense? (:

Reply:
I think those values are quite strict, but also user can configure values on probe side just in case. https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-probes/#configure-probes